gsoc/GSOC_2017_Application_proposal.md

```diff
@@ -0,0 +1,70 @@
+
+Currently I have a a N-component definition idea of the input data to solve the problem. For example, it includes how many changes happens during a specific period of time, how may JSON blobs are changed, what is the time interval when the changes happen, what is the frequency of each change during the monitoring time period etc. All observed data can be reformulated to be the same structure input data. And each component corresponds a weight. Different analyst may have different opinions of the weights distribution. But this kind of input data structure can build a quantized framework of the input data to reflect the changes. The analysts can have their own ways to think about which part is the required significant change. And the good thing is that this kind of data structure can help a unifed way to compare different websites significant changes by using the same weights. 
+
+**Milestones/Timeline:**  
+Week1: Processing and observing data, give a statistical analysis of the samples and formulate the input data’s structure.
+Week2-3: Based on different targets of monitoring websites, optimize the input data’s structure for specific observing targets, mimic the possible the weights of each component and prepare for adjustment.
+Week4-5: Adjust and improve the current machine learning model using small samples and data sets, try to give a recommending system strategy to show the significant changes from different perspectives.
+Week6-7: Keep revising and optimizing the machine learning model and the input data structure using bigger dataset and more different websites, improve its robustness, accuracy and the efficiency.
+Week8-10: Implement this processing module in the real monitoring system, keep optimizing and improving.
+
+**Deliverables:**  
+The final result is a comprehensive input data structure and the processing module with the corresponding machine learning model. If possible, it can have a theoretical framework to deal with such kind of analogous problem.
+
+**Resources:**  
+[List any people, documentation, literature, sample data, or hardware you will need, if applicable]
+I have related knowledge of machine learning theory and algorithms. And I will pay attention to the current issue, record logs and reports, data sampled by Versionista, servers with big storage so that I can have a more intuitive undterstanding of the problem. Besides, all other potential implementations and theories. My advisor Dr. Lijun Chen and other professors, classmates are all my power to solve the problem.
+
+**Setup:**  
+I have already taken the machine learning course and the optimization course. And one project I did was the Expedia hotel recommendation which is very closed to our web monitoring project. Further more, the research I did used optimization a lot. They bring me a good sense about the machine learning algorithm and the optimization strategy. 
```

Comment:
I think "the Expedia Hotel recommendation" assumes that the reader knows what this is. It might be helpful here to include a link or brief explanation of what it is.